packages/input-mask/README.md

```diff
@@ -0,0 +1,82 @@
+
+Primitive that returns an event handler to mask the inputs of an text input element (`<input>`, `<textarea>`) when applied in `oninput` or `onchange`.
+
+## Installation
+
+```bash
+npm install @solid-primitives/input-mask
+# or
+yarn add @solid-primitives/input-mask
+```
+## Usage
+
+The masks come in 3 different formats: function, array and string. There are tools to convert string masks to array masks and array masks to function masks.
+
+```ts
+// string mask:
+// 9 = any number,
+// a = any letter,
+// * = any alphanumeric character
+// any other letter becomes a fixed placeholder
+const isodate = '9999-99-99';
+// array mask: RegExp to match variable parts, strings for fixed placeholders
+const meetingId = [/\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/, '-', /\d/, /\d/, /\d/];
+// function mask: (value, [start, end]) => [value, [start, end]]
+const meetingIdOrName = (value, selection) => (/^\D/.test(value)
+  ? arrayMaskToFn([/\w+/, /\S*/])
+  : anyMaskToFn('999-999-999'))(value, selection);
+
+// converting string mask to array:
+const maskArrayFromString = stringMaskToArray(maskString);
+const maskFuncFromArray = arrayMaskToFn(maskArray);
+const maskFuncFromStringOrArray = anyMaskToFn(mask);
+
+return <>
+  <label for="changeMask">
+    The mask will only be applied after you leave the field
+  </label>
+  <input type="text" id="changeMask" onchange={createInputMask('99/99/9999')} />
+  <label for="inputMask">
+    The mask will be applied on every single input
+  </label>
+  <input type="text" id="inputMask" oninput={createInputMask('99/99/9999')} />
+</>
+```
+
+In most cases you'll want to use `oninput`.
+
+## FAQ
+
+- **Why not support contenteditable?**<br> Getting accessible cross-browser support for contenteditable elements is difficult and results in bloated code, so it was decided to skip it for now. If there is enough demand, a later version might introduce it.
+- **Why `oninput`/`onchange` instead of e.g. `onkeyup`?**<br> There are a few things happening after keydown, -press and -up, which would result in flickering. Generally, you could use `onblur`, but then you'd attempt to apply the mask even if nothing changed, which just seems unnecessarily wasteful.
+- **Will it work with actual events?**<br> Yes, it will work with composed as well as native events. Solid's composed events are more performant, though.
+- **Is there a server version?**<br> No, since it only creates an event handler that will solely run on the client; it makes no sense to create a server version.
+- **Does this provide any error handling?**<br> There is no error handling, but it should work well together with any form handling library.
+- **Can I turn this off and on again?**<br> You can still wrap the output handler in your own handler to turn it off and on again:
+```jsx
+const [useMask, setUseMask] = createSignal(false);
+const mask = createInputMaskHandler('9999-99-99');
+const onInput = (ev) => useMask() && mask(ev);
+return <input oninput={onInput} />;
```

Comment:
Is it necessary use signal here? It might be easier to understand if it were just a `let`.
```suggestion
let useMask = false
const mask = createInputMaskHandler('9999-99-99');
return <input onInput={e => useMask && mask(e)} />;
```

Reply:
I first thought of a prop. A signal at least conveys that there coudl be change. But you're right, we don't need that.